src/components/Todos/Todos.tsx

```diff
@@ -0,0 +1,156 @@
+import { useEffect, useRef, useState } from 'react';
+import { useTodoContext } from '../context/TodoContext';
+
+export const Todos = () => {
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const [editingTitle, setEditingTitle] = useState('');
+  const renameInputRef = useRef<HTMLInputElement | null>(null);
+  const {
+    toggleTodoStatus,
+    updateTodoTitle,
+    deleteTodo,
+    headerInputRef,
+    filteredTodos,
+  } = useTodoContext();
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (editingTodoId !== null) {
+      if (editingTitle.trim() === '') {
+        deleteTodo(editingTodoId);
+      } else {
+        updateTodoTitle(editingTodoId, editingTitle);
+      }
+    }
```

Comment:
The form submission logic could be refactored to avoid deeply nested if conditions.
```suggestion
    if (editingTodoId === null) {
      return
    }
    
    if (editingTitle.trim() === '') {
      deleteTodo(editingTodoId);
    } else {
      updateTodoTitle(editingTodoId, editingTitle);
    }
```

Reply:
![image](https://github.com/user-attachments/assets/fbb72b2b-666f-457a-994d-370e8b700310)